lib/SparqlBenchmarkRunner.ts

```diff
@@ -195,42 +192,30 @@ export class SparqlBenchmarkRunner {
   /**
    * Check if the SPARQL endpoint is available.
    */
-  public isUp(): Promise<boolean> {
-    const fetcher = new SparqlEndpointFetcher({
-      additionalUrlParams: this.additionalUrlParamsInit,
-    });
+  public async endpointAvailable(): Promise<boolean> {
     let timeoutHandle: NodeJS.Timeout | undefined;
-    const promiseTimeout = new Promise<boolean>(resolve => {
-      timeoutHandle = <any> setTimeout(() => resolve(false), 10_000);
+    const promiseTimeout = new Promise<boolean>((resolve) => {
+      timeoutHandle = setTimeout(() => resolve(false), this.availabilityCheckTimeout);
+    });
+    const promiseFetch = new Promise<boolean>((resolve) => {
+      fetch(this.endpoint, {
+        method: 'HEAD',
+      }).then(respose => resolve(respose.ok)).catch(() => resolve(false));
     });
-    const promiseFetch = fetcher.fetchBindings(this.endpoint, this.upQuery)
-      .then(results => new Promise<boolean>(resolve => {
-        results.on('error', () => {
-          clearTimeout(timeoutHandle);
-          resolve(false);
-        });
-        results.on('data', () => {
-          // Do nothing
-        });
-        results.on('end', () => {
-          clearTimeout(timeoutHandle);
-          resolve(true);
-        });
-      }));
-    return Promise.race([ promiseTimeout, promiseFetch ])
-      .catch(() => false);
+    const available = await Promise.race([ promiseTimeout, promiseFetch ]);
+    clearTimeout(timeoutHandle);
+    return available;
   }
 
   /**
    * Wait until the SPARQL endpoint is available.
    */
-  public async waitUntilUp(): Promise<void> {
-    let counter = 0;
-    while (!await this.isUp()) {
+  public async waitForEndpoint(): Promise<void> {
+    const hrstart = process.hrtime();
+    while (!await this.endpointAvailable()) {
       await this.sleep(1_000);
-      this.log(`\rEndpoint not available yet, waited for ${++counter} seconds...`);
```

Comment:
Is there a particular reason this was removed?
We had it there to indicate to the user that the script is still running, and hasn't frozen.

Reply:
The only reason was that I added timestamps to the default CLI logger, and the `\r` was deleting them. Without the `\r`, there would be a lot of lines printed if the endpoint took a long while to become available. I will add back this line, though, because it does make sense to let the user know that the tool is not frozen. Thank you for the feedback!

Reply:
This logging has been added back now.